check_logs.py

```diff
@@ -54,6 +54,34 @@ def _fetch(title, url, dest):
         sys.exit(1)
 
 
+def verify_build():
+    build = get_build()
+
+    # If the build was neither fail nor pass, we need to fetch the status.json
+    # of the particular build to try and get an updated result. We attempt this
+    # up to 7 times.
+    retries = 0
+    max_retries = 7
+    while retries < max_retries:
+        if build["result"] == "fail" or build["result"] == "pass":
+            break
+
+        if retries:
+            time.sleep(2**retries)
+        retries += 1
+
+        status_json = "status.json"
+        url = build["download_url"] + status_json
+        _fetch("status.json", url, status_json)
+        build = json.load(open(status_json))
```

Comment:
should we be reassigning `build` here? Wont `verify_build` return something different than if `get_build` had simply succeeded the first time? For example, say the initial `get_build` fails, we retry once, and get success. `verify_build` will return status.json rather than one of the objects in build.json. Aren't those different schema?

Reply:
As far as I understand it, the `builds.json` that we have `tuxsuite` generate is just the collection of the individual `status.json` files from each build. Overriding the `build` variable is intentional in that case.

Reply:
So it looks like there might be a few more fields that we get from the `tuxsuite` client (comparing https://github.com/ClangBuiltLinux/continuous-integration2/runs/4453112113?check_suite_focus=true with https://builds.tuxbuild.com/21yu3ofQYT1H3nC9EXGtDpMKQ7y/status.json), like `environment`, `finished_time`, `plan`, `user`, and `user_agent`. However, does that matter? We do not use any of those fields.

Reply:
> the builds.json that we have tuxsuite generate is just the collection of the individual status.json files from each build

oh, it is! I never noticed that!